src/test/java/org/smartregister/converters/TaskConverterTest.java

```diff
@@ -13,19 +13,43 @@
 
 	private static Gson gson = new GsonBuilder().registerTypeAdapter(DateTime.class, new TaskDateTimeTypeConverter())
 			.serializeNulls().create();
 
 	@Test
 	public void testConvertToFihrTask() {
 		org.smartregister.domain.Task task = gson.fromJson(taskJson, org.smartregister.domain.Task.class);
-	    Task fihrTask = TaskConverter.convertTasktoFihrResource(task);
-	    assertNotNull(fihrTask);
-	    assertEquals(fihrTask.getStatus().getValueAsEnumConstant().value(), StringUtils.toRootLowerCase(task.getStatus().name()));
-		//TODO : Add assertion on remaining properties
-	    System.out.println(fihrTask);
+		Task fihrTask = TaskConverter.convertTasktoFihrResource(task);
+		assertNotNull(fihrTask);
+		assertEquals(fihrTask.getStatus().getValueAsEnumConstant().value(),
+				StringUtils.toRootLowerCase(task.getStatus().name()));
+		assertEquals(fihrTask.getIdentifier().get(0).getSystem().getValue(), "identifier");
+		assertEquals(fihrTask.getIdentifier().get(0).getValue().getValue(), task.getIdentifier());
+		assertEquals(fihrTask.getGroupIdentifier().getValue().getValue(), task.getGroupIdentifier());
+		assertEquals(fihrTask.getBusinessStatus().getText().getValue(), task.getBusinessStatus());
+		assertEquals(fihrTask.getLocation().getReference().getValue(), task.getLocation());
+		assertEquals(fihrTask.getFor().getReference().getValue(), task.getForEntity());
+		assertEquals(fihrTask.getFocus().getReference().getValue(), task.getFocus());
+		assertEquals(fihrTask.getDescription().getValue(), task.getDescription());
+		assertEquals(fihrTask.getCode().getText().getValue(), task.getCode());
+		assertEquals(fihrTask.getNote().size(), task.getNotes().size());
+		assertEquals(fihrTask.getRequester().getReference().getValue(), task.getRequester());
+		assertEquals(fihrTask.getOwner().getReference().getValue(), task.getOwner());
+		assertEquals(fihrTask.getStatusReason().getText().getValue(), task.getReasonReference());
+		assertEquals(fihrTask.getAuthoredOn().getValue().toString(), task.getAuthoredOn().toString("yyyy-MM-dd'T'HH:mm'Z'"));
+		assertEquals(fihrTask.getLastModified().getValue().toString(),
+				task.getLastModified().toString("yyyy-MM-dd'T'HH:mm'Z'"));
+		assertEquals(fihrTask.getExecutionPeriod().getStart().getValue().toString(),
+				task.getExecutionStartDate().toString("yyyy-MM-dd'T'HH:mm'Z'"));
+		assertEquals(fihrTask.getExecutionPeriod().getEnd().getValue().toString(),
+				task.getExecutionEndDate().toString("yyyy-MM-dd'T'HH:mm'Z'"));
+		assertEquals(fihrTask.getBasedOn().get(0).getReference().getValue(), task.getPlanIdentifier());
+		assertEquals(fihrTask.getMeta().getVersionId().getValue(), String.valueOf(task.getServerVersion()));
+		assertEquals(fihrTask.getIntent().getValueAsEnumConstant().value(), "plan");
+		//TODO: Do we need to add assertions for Notes individually?
```

Comment:
You can test size of notes and one note is properly converted

Reply:
Done